AAI/AAIConnectProfile.md

```diff
@@ -305,7 +303,7 @@ Payload:
 -   iss: MUST be able to be appended with .well-known/openid-configuration to
     get spec of broker.
 
--   sub: authenticated user unique identifier
+-   sub: authenticated user unique identifier. email is suggested but another identifier MAY be used for privacy reasons.
```

Comment:
I think that email should not be recommended as an unique identifier. It has multiple issues. Users usually use more than one email address. They change emails when they change their names, e.g. when they marry. Email is considered personal data by European GDPR (General Data Protection Regulation), it should not be disclosed without written consent and the user may require delete of it from all systems. 

You may want to google up the long discussions about eduPersonPrincipalName, eduPersonTargetedId, eduPersonUniqueId and samlSubjectID or read the "Motivation" part of http://docs.oasis-open.org/security/saml-subject-id-attr/v1.0/csprd03/saml-subject-id-attr-v1.0-csprd03.pdf

Reply:
So you don't think it's even worth the "suggestion" that email is to be used? That's reasonable. It's PII here in the States and it'll be part of whatever GDPR-like legislation eventually happens. It's already part of NIST-800-53 Appendix j (which is similar to GDPR). 

That said, what do YOU recommend we use for Sub as there will be no central authority handing out IDs.

Reply:
I know ELIXIR has an obfuscated sub but it's centralized. In a non-centralized approach, as we have, we need to have "SOME" sub. Maybe we just accept that 'subs' can clash as a risk and advise that implementers make their own risk acceptance to trust which Identity Providers/Brokers they allow access_tokens from? But that seems that it's maybe not a great solution...

Reply:
My suggestion is to take the ELIXIR approach, it means a randomly generated string with scope ("@somedomain.com") at the end.